Flaky Test Case: TestValidOAuthAccessToken

I see this test case fail from time to time.. 

```
--- FAIL: TestValidOAuthAccessToken (0.00s)
	assertions.go:241: 
	Error Trace:	service_whitebox_test.go:83
	Error:		Expected nil, but got: &url.Error{Op:"Get", URL:"https://api.github.com/user/emails", Err:(*errors.errorString)(0xc8203c7970)}

	assertions.go:241: 
	Error Trace:	service_whitebox_test.go:84
	Error:		Should NOT be empty, but was []
```

--- a/login/service_whitebox_test.go
+++ b/login/service_whitebox_test.go
@@ -3,6 +3,7 @@ package login
 import (
 	"fmt"
 	"testing"
+	"time"
 
 	"github.com/almighty/almighty-core/account"
 	"github.com/almighty/almighty-core/configuration"
@@ -66,10 +67,23 @@ func TestValidOAuthAccessToken(t *testing.T) {
 		AccessToken: configuration.GetGithubAuthToken(),
 		TokenType:   "Bearer",
 	}
-
 	emails, err := loginService.getUserEmails(context.Background(), accessToken)
-	assert.Nil(t, err)
-	assert.NotEmpty(t, emails)
+	var maxtries int // Number of tries to reach GitHub
+	if err != nil {
+		for maxtries = 0; maxtries < 10; maxtries++ {
+			time.Sleep(5 * time.Second) // Pause before the next retry
+			emails, err = loginService.getUserEmails(context.Background(), accessToken)
+			if err == nil {
+				assert.Nil(t, err)
+				assert.NotEmpty(t, emails)
+				break
+			}
+		}
+	}
+	if maxtries == 10 {
+		t.Error("Test failed, Maximum Retry limit reached", err) // Test failed after trial for 10 times
+	}
+
 }
 
 func TestInvalidOAuthAccessToken(t *testing.T) {

--- a/login/service_whitebox_test.go
+++ b/login/service_whitebox_test.go
@@ -73,20 +73,17 @@ func TestValidOAuthAccessToken(t *testing.T) {
 		for trials = 0; trials < 10; trials++ {
 			emails, err = loginService.getUserEmails(context.Background(), accessToken)
 			if err == nil {
-				assert.Nil(t, err)
 				assert.NotEmpty(t, emails)
 				break
 			}
 			time.Sleep(5 * time.Second) // Pause before the next retry
 		}
+		if trials == 10 {
+			t.Error("Test failed, Maximum Retry limit reached", err) // Test failed after trial for 10 times
+		}
 	} else {
-		assert.Nil(t, err)
 		assert.NotEmpty(t, emails)
 	}
-	if trials == 10 {
-		t.Error("Test failed, Maximum Retry limit reached", err) // Test failed after trial for 10 times
-	}
-
 }
 
 func TestInvalidOAuthAccessToken(t *testing.T) {

--- a/login/service_whitebox_test.go
+++ b/login/service_whitebox_test.go
@@ -68,19 +68,22 @@ func TestValidOAuthAccessToken(t *testing.T) {
 		TokenType:   "Bearer",
 	}
 	emails, err := loginService.getUserEmails(context.Background(), accessToken)
-	var maxtries int // Number of tries to reach GitHub
+	var trials int // Number of tries to reach GitHub
 	if err != nil {
-		for maxtries = 0; maxtries < 10; maxtries++ {
-			time.Sleep(5 * time.Second) // Pause before the next retry
+		for trials = 0; trials < 10; trials++ {
 			emails, err = loginService.getUserEmails(context.Background(), accessToken)
 			if err == nil {
 				assert.Nil(t, err)
 				assert.NotEmpty(t, emails)
 				break
 			}
+			time.Sleep(5 * time.Second) // Pause before the next retry
 		}
+	} else {
+		assert.Nil(t, err)
+		assert.NotEmpty(t, emails)
 	}
-	if maxtries == 10 {
+	if trials == 10 {
 		t.Error("Test failed, Maximum Retry limit reached", err) // Test failed after trial for 10 times
 	}
 

